internal/pkg/keeper/client.go

```diff
@@ -0,0 +1,283 @@
+//
+// Copyright (C) 2022 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package keeper
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-registry/v2/pkg/dtos"
+	"github.com/edgexfoundry/go-mod-registry/v2/pkg/types"
+)
+
+type keeperClient struct {
+	config              *types.Config
+	keeperUrl           string
+	serviceKey          string
+	serviceAddress      string
+	servicePort         int
+	healthCheckRoute    string
+	healthCheckInterval string
+}
+
+func NewKeeperClient(registryConfig types.Config) (*keeperClient, error) {
+	client := keeperClient{
+		config:     &registryConfig,
+		serviceKey: registryConfig.ServiceKey,
+		keeperUrl:  registryConfig.GetRegistryUrl(),
+	}
+
+	// ServiceHost will be empty when client isn't registering the service
+	if registryConfig.ServiceHost != "" {
+		client.servicePort = registryConfig.ServicePort
+		client.serviceAddress = registryConfig.ServiceHost
+		client.healthCheckRoute = registryConfig.CheckRoute
+		client.healthCheckInterval = registryConfig.CheckInterval
+	}
+
+	return &client, nil
+}
+
+func (k *keeperClient) Register() error {
+	if k.serviceKey == "" || k.serviceAddress == "" || k.servicePort == 0 ||
+		k.healthCheckRoute == "" || k.healthCheckInterval == "" {
+		return fmt.Errorf("unable to register service with keeper: Service information not set")
+	}
+
+	registrationReq := dtos.AddRegistrationRequest{
+		BaseRequest: dtos.BaseRequest{
+			Versionable: dtos.Versionable{ApiVersion: ApiVersion},
+		},
+		Registration: dtos.RegistrationDTO{
+			ServiceId: k.serviceKey,
+			Host:      k.serviceAddress,
+			Port:      k.servicePort,
+			HealthCheck: dtos.HealthCheck{
+				Interval: k.healthCheckInterval,
+				Path:     k.healthCheckRoute,
+				Type:     "http",
+			},
+		},
+	}
+
+	jsonEncodedData, err := json.Marshal(registrationReq)
+	if err != nil {
+		return fmt.Errorf("failed to encode registration request: %s", err.Error())
+	}
+
+	req, err := http.NewRequest(http.MethodPost, k.config.GetRegistryUrl()+ApiRegisterRoute, bytes.NewReader(jsonEncodedData))
+	if err != nil {
+		return fmt.Errorf("failed to create register request: %s", err.Error())
+	}
+	req.Header.Set(ContentType, ContentTypeJSON)
+
+	client := http.Client{Timeout: 10 * time.Second}
```

Comment:
declare the timeout value as a const, because we use it several times in this file